app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepository.kt

```diff
@@ -0,0 +1,18 @@
+package space.taran.arkmemo.data.repositories
+
+import android.content.Context
+import space.taran.arkmemo.files.text.TextNoteFiles
+import space.taran.arkmemo.models.TextNote
+import javax.inject.Inject
+
+class TextNotesRepository @Inject constructor() {
+
+    fun saveNote(context: Context, note: TextNote) =
+        TextNoteFiles.saveNote(context, note)
+
+    fun deleteTextNote(context: Context, note: TextNote) =
+        TextNoteFiles.deleteNote(context, note)
+
+    fun getAllTextNotes(context: Context) =
+        TextNoteFiles.readAllNotes(context)
```

Comment:
`TextNoteFiles` really needed?
Maybe remove it and move all code here?

Also, you can get application `Context` from di and not pass through args.

Reply:
> Also, you can get application Context from di and not pass through args.

@ShubertMunthali yeah, better to do it using DI. At the moment, I see several files with `context: Context` argument in methods.